lxd/metrics/metrics.go

```diff
@@ -24,26 +22,19 @@ func NewMetricSet(labels map[string]string) *MetricSet {
 	return &out
 }
 
-// FilterSamples filters the existing MetricSet using the given permission checker. Samples not containing the "project" label are skipped.
-func (m *MetricSet) FilterSamples(permissionChecker func(entityURL *api.URL) bool) {
+// FilterSamples filters the existing MetricSet using the given permission checker.
+func (m *MetricSet) FilterSamples(permissionChecks []func(labels map[string]string) bool) {
 	for metricType, samples := range m.set {
 		allowedSamples := make([]Sample, 0, len(samples))
 		for _, s := range samples {
-			projectName := s.Labels["project"]
-			instanceName := s.Labels["name"]
-			if projectName == "" {
-				continue
-			}
-
-			var hasPermission bool
-
-			if instanceName != "" {
-				hasPermission = permissionChecker(entity.InstanceURL(projectName, instanceName))
-			} else {
-				hasPermission = permissionChecker(entity.ProjectURL(projectName))
+			permittedSample := false
+			for _, permissionCheck := range permissionChecks {
+				if permissionCheck(s.Labels) {
+					permittedSample = true
```

Comment:
Should we break here and not try any other permission checkers that may have been supplied?

Reply:
Thanks, fixed.